imageroot/actions/configure-module/11validate_mail_module

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (C) 2022 Nethesis S.r.l.
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+
+import json
+import sys
+import os
+import agent
+
+agent.set_weight(os.path.basename(__file__), 0) # Validation step, no task progress at all
+
+# Try to parse the stdin as JSON.
+# If parsing fails, output everything to stderr
+data = json.load(sys.stdin)
+
+agent_id = os.getenv("AGENT_ID", "")
+if not agent_id:
+    raise Exception("AGENT_ID not found inside the environemnt")
+
+# Connect to redis
+rdb = agent.redis_connect()
+
+if "mail_module" in data and data["mail_module"] != os.getenv("MAIL_MODULE"):
+    mail_module = data["mail_module"]
+    if not rdb.exists(f"module/{mail_module}"):
+            agent.set_status('validation-failed')
+            json.dump([{'field':'mail_module','parameter':'mail_module','value': data["mail_module"],'error':'mail_module_is_not_valid'}], fp=sys.stdout)
+            sys.exit(2)
```

Comment:
IIRC the condition here is never true because that key does not exist.

```suggestion
```

The next check is correct and should be enough for us.